### PR TITLE
ShlomiShitrit/issue29 - null values

### PR DESCRIPTION
**Pull request by ShlomiShitrit**
This pull request adds a new feature to fill null values in rows. It includes changes to the `Column` class, `DataType` type, and `fillNullValuesParams` type. The `Column` class now has a `fillNullValues` method that accepts different methods such as "mean", "median", "mode", "remove", or "custom" to fill null values with appropriate values. The `fillNullValues` method can be used on individual columns or on all columns at once using the `fillNullValuesAll` method. This feature enhances data cleaning and preprocessing capabilities.

**Example of use case:**

First the developer needs to define what is considered as null values. This is the values that will get filled:

image

Then the developer can either change a specific column:

image

Or the entire csv:

image

### CODE REVIEW

File: scripts/demo.ts [hunk 1]
```diff
@@ -62,7 +62,23 @@ dbBot.slots = slots;
...
 ///////////////////
 
-dbBot.loadFile("./pokemon.csv");
+dbBot.loadFile("./sw_characters.csv");
+
+/////////////////////////
+// null values in rows //
+/////////////////////////
+
+const nullValues = [null, "NA", NaN];
```
Comment (nadavWeisler) [inline scripts/demo.ts:71]: In the future this should be a solid demo, and will not be changed for every tryout

File: DBbot/classes/column.ts [hunk 2]
```diff
@@ -58,4 +58,87 @@ export class Column {
...
+
+    private fillRow(inputValue: NumOrStr, nullValue: any[] = [null]): void {
+        this.rows.forEach((row, index) => {
+            if (nullValue.includes(row)) {
+                this.rows[index] = inputValue;
+            }
+        });
+    }
+
+    public fillNullValues(
```
Comment (nadavWeisler) [inline DBbot/classes/column.ts:120]: elegant, good work!

File: DBbot/classes/column.ts [hunk 3]
```diff
@@ -58,4 +58,87 @@ export class Column {
...
+        }, 0);
+        const mean = sum / this.rows.length;
+        return Math.round(mean * 100) / 100;
+    }
+
+    public mode(): number {
+        if (this.dataType === "string") {
+            throw new Error("Cannot calculate mode for string data type");
+        }
+        const counts = this.rows.reduce((acc, curr) => {
```
Comment (nadavWeisler) [inline DBbot/classes/column.ts:78]: What is acc and curr? try to not use shortcuts in variable names

File: DBbot/classes/column.ts [hunk 4]
```diff
@@ -58,4 +58,87 @@ export class Column {
     public addOperator(operator: Operator): void {
         this.operatorsArray.push(operator);
     }
+
+    public mean(): number {
+        if (this.dataType === "string") {
+            throw new Error("Cannot calculate mean for string data type");
```
Comment (nadavWeisler) [inline DBbot/classes/column.ts:64]: I saw all the errors are hard-coded; in ideal words its shoud be in a resources file; I think here it is fine, but I just notice it

File: DBbot/classes/column.ts (hunk 3 above)
Comment (ShlomiShitrit) [reply DBbot/classes/column.ts:82]: fixed

File: DBbot/classes/column.ts (hunk 4 above)
Comment (ShlomiShitrit) [reply DBbot/classes/column.ts:68]: fixed

**Pull request merged**